### ISSUE TITLE
Bring back rule S6601: Boolean expressions should be safe

### ISSUE THREAD

**Issue by ilia-kebets-sonarsource** [SonarSource/SonarJS#4471]
We have [implemented the rule](https://github.com/SonarSource/SonarJS/pull/3859) decorating [typescript-eslint's strict-boolean-expressions](https://typescript-eslint.io/rules/strict-boolean-expressions), but decided to [drop it because it was noisy](https://github.com/SonarSource/SonarJS/pull/3867).

But it might be useful to get back to it, maybe rephrase it's goal as to make code more intentional.

The SQ frontend team has a [custom rule](https://github.com/SonarSource/sonar-enterprise/blob/master/server/sonar-web/eslint-local-rules/no-implicit-coercion.js) for it.

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-48
